drivers/adc/adc_stm32.c

```diff
@@ -706,6 +840,16 @@ static int start_read(const struct device *dev,
 	}
 #endif
 
+#if defined(CONFIG_ADC_STM32_DMA)
+	/* The buffer needs to be 32 byte aligned for the cache invalidation
+	 * function to operate properly, using __aligned(32)
+	 */
+	if (!IS_PTR_ALIGNED_BYTES(data->buffer, 32)) {
+		LOG_ERR("The buffer needs to be 32-byte aligned");
```

Comment:
Is this check only for making sure that `SCB_InvalidateDCache_by_Addr()` gets a 32-byte aligned address? The version of that function used in the current CMSIS properly handles unaligned addresses: https://github.com/zephyrproject-rtos/cmsis/blob/master/CMSIS/Core/Include/cachel1_armv7.h#L320 *D-Cache memory blocks which are part of given address + given size are cleaned.*

There are other variants of the documentation for that function floating about which claim it needs 32-byte alignment, but the one used in Zephyr doesn't.

Reply:
You are indeed correct. I was looking at the documentation and not the function body. I also verified that it still works by offsetting the buffers by one, and all tests still passed. So I'm removing this check.